panther_hardware_interfaces/src/gpio_controller.cpp

```diff
@@ -0,0 +1,229 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <thread>
+#include <utility>
+
+#include "gpiod.hpp"
+
+#include "panther_hardware_interfaces/gpio_controller.hpp"
+
+namespace panther_hardware_interfaces
+{
+
+Watchdog::Watchdog(std::shared_ptr<panther_gpiod::GPIODriver> gpio_driver)
+: gpio_driver_(std::move(gpio_driver))
+{
+  if (!gpio_driver_->IsPinAvaible(watchdog_pin_)) {
+    throw std::runtime_error("Watchdog pin is not configured.");
+  }
+}
+
+Watchdog::~Watchdog() { TurnOff(); }
+
+bool Watchdog::TurnOn()
+{
+  if (IsWatchdogEnabled()) {
+    return true;
+  }
+
+  if (!gpio_driver_->IsPinAvaible(watchdog_pin_)) {
+    throw std::runtime_error("Watchdog pin is not configured.");
+  }
+
+  watchdog_thread_enabled_ = true;
+  watchdog_thread_ = std::make_unique<std::thread>(&Watchdog::WatchdogThread, this);
+
+  return IsWatchdogEnabled();
+}
+
+bool Watchdog::TurnOff()
+{
+  if (!IsWatchdogEnabled()) {
+    return true;
+  }
+
+  watchdog_thread_enabled_ = false;
+  watchdog_thread_->join();
+
+  return !IsWatchdogEnabled();
+}
+
+void Watchdog::WatchdogThread()
+{
+  while (watchdog_thread_enabled_) {
+    bool value = gpio_driver_->IsPinActive(watchdog_pin_);
+
+    gpio_driver_->SetPinValue(watchdog_pin_, !value);
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+  }
+
+  gpio_driver_->SetPinValue(watchdog_pin_, false);
+}
+
+bool Watchdog::IsWatchdogEnabled() const
+{
+  return watchdog_thread_ && watchdog_thread_->joinable();
+}
+
+void GPIOControllerInterface::ConfigureGpioStateCallback(
+  const std::function<void(const panther_gpiod::GPIOInfo &)> & callback)
+{
+  if (!gpio_driver_) {
+    throw std::runtime_error("GPIO driver has not been initialized yet.");
+  }
+
+  gpio_driver_->ConfigureEdgeEventCallback(callback);
+}
+
+bool GPIOControllerInterface::IsPinActive(const panther_gpiod::GPIOPin pin) const
+{
+  return gpio_driver_->IsPinActive(pin);
+}
+
+bool GPIOControllerInterface::IsPinAvaible(const panther_gpiod::GPIOPin pin) const
+{
+  return gpio_driver_->IsPinAvaible(pin);
+}
+
+void GPIOControllerPTH12X::Start()
+{
+  gpio_driver_ = std::make_shared<panther_gpiod::GPIODriver>(gpio_config_info_storage_);
+  gpio_driver_->GPIOMonitorEnable(true, 60);
+
+  gpio_driver_->SetPinValue(panther_gpiod::GPIOPin::VMOT_ON, true);
+  MotorsEnable(true);
+
+  watchdog_ = std::make_unique<Watchdog>(gpio_driver_);
+}
+
+// TODO: void
+bool GPIOControllerPTH12X::EStopTrigger()
+{
+  if (!watchdog_->TurnOff()) {
+    throw std::runtime_error("Can't stop watchdg thread");
+  }
+  return true;
+}
+
+bool GPIOControllerPTH12X::EStopReset()
+{
+  auto e_stop_pin = panther_gpiod::GPIOPin::E_STOP_RESET;
```

Comment:
```suggestion
  using panther_gpiod::GPIOPin;
```
and change e_stop_pin to panther_gpiod::E_STOP_RESET. This is just a suggestion, if you prefer variable, then add `const`.

Reply:
I think `const` is the clearest option, okay?